src/main/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/PgEventBookings.java

```diff
@@ -82,22 +78,27 @@ public EventBooking add(final String eventId, final Long userId, final BookingSt
         try (Connection conn = ds.getDatabaseConnection()) {
             Date creationDate = new Date();
             pst = conn.prepareStatement(
-                    "INSERT INTO event_bookings (id, user_id, event_id, status, created, updated, additional_booking_information) VALUES (DEFAULT, ?, ?, ?, ?, ?, ?::text::jsonb)",
+                    "INSERT INTO event_bookings (id, user_id, reserved_by, event_id, status, created, updated, additional_booking_information) VALUES (DEFAULT, ?, ?, ?, ?, ?, ?, ?::text::jsonb)",
                     Statement.RETURN_GENERATED_KEYS);
             pst.setLong(1, userId);
-            pst.setString(2, eventId);
-            pst.setString(3, status.name());
-            pst.setTimestamp(4, new java.sql.Timestamp(creationDate.getTime()));
+            try {
+                pst.setLong(2, reserveById);
+            } catch (NullPointerException e) {
+                pst.setNull(2, Types.INTEGER);
```

Comment:
If we made it a foreign key and always populated it with the user who made the booking / reservation we wouldn't need nasty catches for null pointers.